executors/cpp/localedisplaynames.cpp

```diff
@@ -0,0 +1,76 @@
+/********************************************************************
+ * testing icu4c for locale display names
+ */
+
+#include <json-c/json.h>
+
+#include <unicode/utypes.h>
+
+#include <unicode/locdspnm.h>
+#include <unicode/uldnames.h>
+#include <unicode/unistr.h>
+
+#include <string>
+#include <cstring>
+
+#include "./util.h"
+
+using icu::Locale;
+using icu::UnicodeString;
+using icu::LocaleDisplayNames;
+
+const string TestLocaleDisplayNames (json_object *json_in) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  json_object *label_obj = json_object_object_get(json_in, "label");
+  string label_string = json_object_get_string(label_obj);
+
+  // The locale in which the name is given.
+  json_object *locale_label_obj =
+      json_object_object_get(json_in, "locale_label");
+  string locale_string = json_object_get_string(locale_label_obj);
+
+  // The locales's name to be displayed.
+  json_object *language_label_obj = json_object_object_get(
+      json_in, "language_label");
+  string language_label_string = json_object_get_string(language_label_obj);
+
+  // Either standard or dialect names for the locale.
+  json_object *language_display_obj = json_object_object_get(
+      json_in, "languageDisplay");
+  string language_display_string = json_object_get_string(language_display_obj);
```

Comment:
rename these local variables from `language_*` to `locale_*`? you did above, just didn't continue that renaming here.

Reply:
Done